e2e_test/source_inline/kafka/protobuf/recover.slt

```diff
@@ -0,0 +1,97 @@
+system ok
+rpk topic create 'test-pb-struct'
+
+
+system ok
+jq -sR '{"schema":.,"schemaType":"PROTOBUF"}' << EOF | curl -X POST -H 'content-type: application/json' -d @- 'http://schemaregistry:8082/subjects/test-pb-struct-value/versions'
+syntax = "proto3";
+package test;
+message User {
+  int32 id = 1;
+  Name name = 2;
+}
+message Name {
+  string first_name = 1;
+  string last_name = 2;
+}
+EOF
+
+
+# create a source with v1 schema
+statement ok
+create source s with (
+  connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+  topic = 'test-pb-struct')
+format plain encode protobuf (
+  schema.registry = 'http://schemaregistry:8082',
+  message = 'test.User');
+
+
+# register a v2 schema
+system ok
+jq -sR '{"schema":.,"schemaType":"PROTOBUF"}' << EOF | curl -X POST -H 'content-type: application/json' -d @- 'http://schemaregistry:8082/subjects/test-pb-struct-value/versions'
+syntax = "proto3";
+package test;
+message User {
+  int32 id = 1;
+  Name name = 2;
+}
+message Name {
+  string first_name = 1;
+  string last_name = 2;
+  string middle_name = 3;
+}
+EOF
+
+
+# trigger recovery
+statement ok
+recover;
+
+
+sleep 2s
+
+
+# produce a v2 message
+statement ok
+create sink sk as select
+  1 as id,
+  row('Alan', 'Turing', 'Mathison')::struct<first_name varchar, last_name varchar, middle_name varchar> as name
+with (
+  connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+  topic = 'test-pb-struct')
+format plain encode protobuf (
+  schema.registry = 'http://schemaregistry:8082',
+  message = 'test.User');
+
+
+sleep 1s
+
+
+# reading as v1 shall not panic
+query IT
+select * from s;
+----
+1 (Alan,Turing)
```

Comment:
The panic reproduced on first commit of this PR:
https://buildkite.com/risingwavelabs/pull-request/builds/57644#0191dac4-4ad6-42dc-8f8c-2c3e4b182ed5
```
failed to run `e2e_test/source_inline/kafka/protobuf/recover.slt`
Caused by:
    query failed: db error: ERROR: Failed to run the query
    Caused by:
      execution panic
    [SQL] select * from s;
    at e2e_test/source_inline/kafka/protobuf/recover.slt:74
```